### PR TITLE
284 stream upload

### PR DESCRIPTION
**Pull request by kindly**
#284

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 django-bootstrap3==7.0.1
-requests==2.10.0
+requests==2.11.1
 django-environ==0.4.0
-raven==5.23.0
-elasticsearch==2.3.0
--e git+https://github.com/OpenDataServices/flatten-tool.git@dceac82323271c4cc5956275481d762d81976eaf#egg=flattentool
+raven==5.24.3
+elasticsearch==2.4.0
+-e git+https://github.com/OpenDataServices/flatten-tool.git@dceac82323271c4cc5956275481d762d81976eaf#egg=flattentool-dev
```
Comment (robredpath) [inline requirements.txt]: Is this some reqs weirdness?

Comment (Bjwebb) [inline requirements.txt]: Yes, it's updated everything to the latest version. Reverting the two .txt files and running `./update_requirements --new-only` should fix this https://github.com/OpenDataServices/grantnav#adding-and-updating-requirements (or we could choose to have requirements updated)

**Comment by Bjwebb** [comment]
This still uses ~ 1 GB of memory for me.

I wonder if we should be using `streaming_bulk` instead? http://elasticsearch-py.readthedocs.io/en/master/helpers.html#elasticsearch.helpers.streaming_bulk

Also, now that we have a [package schema](http://www.threesixtygiving.org/standard/reference/#toc-360giving-json-schemas) it's probably safer now to assume `grants_key = "grants"` instead of trying to guess it.

**Comment by Bjwebb** [comment]
Although, checking a data load with the previous code, that gets closer to 2GB, so this could still be worth merging.

**Comment by Bjwebb** [comment]
I've modified the grants key behaviour to always assume "grants". @kindly could you check that looks alright?

I'm happy for this to get merged, because it does seem to be an improvement.

**Comment by Bjwebb** [comment]
> I wonder if we should be using streaming_bulk instead? http://elasticsearch-py.readthedocs.io/en/master/helpers.html#elasticsearch.helpers.streaming_bulk

Looks like `streaming` calls `streaming_bulk` under the hood, so this isn't relevant.

**Pull request merged**